### DISCUSSION TITLE
error[E0425]: cannot find function, tuple struct or tuple variant `Ok` in this scope

### DISCUSSION THREAD

**Discussion by smz2025** [BurntSushi/ripgrep#2743] [Q&A] [answered]
Hi,
I am trying to install ripgrep with latest cargo 1.76.0. FYI, i am new to cargo.
I ran "cargo install ripgrep" but it returned bunch of messages with regards to Some function.

BTW before this i also run with CARGO_TARGET_DIR set to /tmp/cargo-installUAGC4O, both same error.    Please advise what I need to do. Thank you in advance.

this is some extract of the error messages:
error[E0425]: cannot find function, tuple struct or tuple variant `Some` in this scope
   --> /tmp/abc/ripgrep-14.1.0/.cargo/registry/src/index.crates.io-6f17d22bba15001f/regex-syntax-0.8.2/src/utf8.rs:398:13
    |
398 |             Some((
    |             ^^^^ not found in this scope

error[E0425]: cannot find function, tuple struct or tuple variant `Some` in this scope
   --> /tmp/abc/ripgrep-14.1.0/.cargo/registry/src/index.crates.io-6f17d22bba15001f/regex-syntax-0.8.2/src/utf8.rs:418:13
    |
418 |             Some(Utf8Range::new(start, end))
    |             ^^^^ not found in this scope

Some errors have detailed explanations: E0405, E0408, E0412, E0416, E0425, E0463, E0531.
error: could not compile `regex-syntax` (lib) due to 1320 previous errors
error: failed to compile `ripgrep v14.1.0`, intermediate artifacts can be found at `/tmp/cargo-installUAGC4O`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.

**Comment by BurntSushi** [comment, marked as answer]
This looks to me like something is wrong with your Rust/Cargo installation. The errors are telling you that it can't find the `Some` function, which means `core` isn't available. But you might consider trying to compile some other Rust programs to see if you get similar errors. My hypothesis is that you will.

Like how did you install Rust?  Are you cross compiling? Which target are you on?

**Reply by smz2025** [reply]
I downloaded the build version from the website itself; https://static.rust-lang.org/dist/rust-1.76.0-x86_64-unknown-linux-gnu.tar.gz
I tried to build from source but I hit a roadblock at rustc-driver it complained about libstd so I took the easy way above.

I also tried version 1.72.0 but hit the same error.  I am on SLES15SP4 operating system.

**Reply by BurntSushi** [reply]
Do you mean SUSE Linux?

Sorry, but I don't know what the problem is. I've seen these errors before, but they are always in the context of having a bunk Rust installation or not setting up the right cross compilation environment. This is definitely not a ripgrep problem. I'm sorry, but I don't know how to help you. I'd encourage you to do a little more debugging on your own, such as trying to compile other Rust code. You might have luck seeking help in a forum geared toward helping out with more basic problems. There's a Discord for example. And an official [Rust users forum](https://users.rust-lang.org/).
